HowTo/use_gRPC_with_dotnet.md

```diff
@@ -0,0 +1,122 @@
+
+The following tutorial builds from the created gRPC client and server in that documentation.  If you already have a gRPC client and server, you may use these steps to add to existing .NET apps as well.
+
+### Setup the gRPC Server app
+In order to prepare our gRPC server application to deploy to App Service, we will need to configure the kestrel to listen to an additional port that only listens for HTTP/2
+
+In your **Program.cs** add the following code to configure the kestrel.  In this example we're listening to port 8585, but you can use another number.
+
+```C#
+// Configure Kestral to listen on a specific HTTP port 
+builder.WebHost.ConfigureKestrel(options => 
+{ 
+    options.ListenAnyIP(8080); 
+    options.ListenAnyIP(8585, listenOptions => 
+    { 
+        listenOptions.Protocols = Microsoft.AspNetCore.Server.Kestrel.Core.HttpProtocols.Http2; 
+    }); 
+});
+
+```
+
+Once configured this will ensure that your application is listening to a specific HTTP/2 only port, which will be needed when we deploy to App Service.
+
+If you are using the templated code from the ASP.NET tutorial we will need to remove the EndpointDefaults configuration in the appsettings.json file.  In the **appsettings.json** file, remove the commented code below.
+
+```json
+{ 
+  "Logging": { 
+    "LogLevel": { 
+      "Default": "Information", 
+      "Microsoft.AspNetCore": "Warning" 
+    } 
+  }, 
+  "AllowedHosts": "*", 
+  //  "EndpointDefaults": { 
+  //    "Protocols": "Http2" 
+  //  } 
+  //} 
+}
+
+```
+We no longer need this code since we have configured the protocol options in our Program.cs file and it will cause an error if deployed.  Once this is done your application is now ready to deploy to App Service.
+
+### Deploy to App Service
+Now that you have your server application setup and running locally, you can go to the portal and create your web app.  One thing to note is that gRPC is currently only supported on Linux so be sure to choose this option when creating your web app.
+
+Create your web app as you normally would.  Choose **Code** as your Publish option.  Choose **.NET 6 (LTS)** as your Runtime stack and **Linux** as your Operating System.  
+
+Now that your web app is created, you'll need to do the following before deploying your application:
+
+#### 1. Enable HTTP version
+The first setting you'll need to configure is the HTTP version
+1. Navigate to **Configuration** under **Settings** in the left pane of your web app
+2. Click on the **General Settings** tab and scroll down to **Platform settings**
+3. Go to the **HTTP version** drop-down and select **2.0**
+4. Click **save**
+
+This will restart your application and configure your site to allow clients to connect over HTTP/2.
```

Comment:
Technically, clients still can't connect to the _site_ over HTTP2 at this step. Only to the FrontEnd.

Reply:
Would this work as a replacement?
"This will restart your application and configure the Front End to allow clients to make HTTP/2 calls."

Reply:
Sure that would work (as long as updating the config actually restarts the site, I am not familiar with this behavior)

Reply:
It should be correct.  After you hit save in the Portal, a pop up tells you any changes to the application settings will cause the app to restart.